tekton/resources/checks/check-pr-has-kind-label.yaml

```diff
@@ -0,0 +1,203 @@
+  - apiVersion: tekton.dev/v1alpha1
+    kind: TaskRun
+    metadata:
+      name: check-pr-labels-$(uid)
+    spec:
+      taskSpec:
+        inputs:
+          resources:
+          - name: source
+            type: git
+          - name: pr
+            type: pullRequest
+          params:
+          - name: labels 
+            description: The labels attached to the Pull Request
+          - name: gitRepository
+            description: The URL of the git repository
+          - name: pullRequestUrl
+            description: The URL of the Pull Request
+        outputs:
+          resources:
+          - name: pr
+            type: pullRequest
+        steps:
+        - name: copy-pr-to-output
+          image: busybox
+          script: |
+            #!/bin/sh
+            mkdir -p $(outputs.resources.pr.path)
+            cp -r $(inputs.resources.pr.path)/* $(outputs.resources.pr.path)/
+        - name: check-labels
+          env:
+          - name: GH_LABELS
+            value: '$(params.labels)'
+          image: python:3-alpine
+          script: |
+            #!/usr/bin/env python
+            import json
+
+            labels = json.loads('$(params.labels)')
+            labelNames = list(map(lambda e: e["name"], labels))
+            kindLabels = list(filter(lambda e: "kind" in e, labelNames))
+            comment_template=""
+            if (len(kindLabels) > 1 or len(kindLabels) == 0):
+              print('failed')
+              comment_template += """
+            **This PR cannot be merged:** expecting one kind/ label
+
+            <details>
+            The following `kind/` labels are available:</p>
+
+            These are some kind labels
+
+            </details>
+            """
+              new_comment_path = "$(outputs.resources.pr.path)/comments/new.json"
+              comment_body = dict(body=comment_template)
+              with open(new_comment_path, "w") as comment:
+                json.dump(comment_body, comment)
+
+              fail_check = {
+                "State": "failure",
+                "Label": "check-pr-has-kind-label",
+                "Desc": "The PR requires a kind/ label to merge"
+              }
+
+              new_check_path = "$(outputs.resources.pr.path)/status/check-pr-has-kind-label.json"
+              with open(new_check_path, "w") as check:
+                json.dump(fail_check, check)
+            else:
+              print('success')
+              success_check = {
+                "State": "success",
+                "Label": "check-pr-has-kind-label",
+                "Desc": "This PR has a kind/ label"
+              }
+              new_check_path = "$(outputs.resources.pr.path)/status/check-pr-has-kind-label.json"
+              with open(new_check_path, "w") as check:
+                json.dump(success_check, check)
+        - name: cat-check
+          image: ubuntu
+          script: |
+             #!/bin/bash
+             cat $(outputs.resources.pr.path)/status/*
```

Comment:
This is for debug purposes only? 
Do we need to log *all* checks?

Reply:
Correct, this is for debugging

Reply:
Removed